File: algorithm/507. Perfect Number/507.cpp
class Solution {
public:
    bool checkPerfectNumber(int num) {
        if (num==1) return 0;
        int result=1,tem=2;
        while(tem*tem<=num)
        {
            if(num%tem==0)
            {
                result+=tem;
                if(num/tem!=tem)
                    result+=num/tem;
            }
            tem++;
        }
        return result==num;
    }
};

File: algorithm/507. Perfect Number/507.py
class Solution(object):
    def checkPerfectNumber(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num==1:
            return False
        tem,result=2,1
        while tem*tem<=num:
            if num%tem==0:
                result+=tem
                if num/tem!=tem:
                    result+=num/tem
            tem+=1
        return result==num
